### PR TITLE
Fix non-tar bz2 extractions

### PR DESCRIPTION
**Pull request by egibs**
I noticed a few extraction errors for `.bz2` files:
CleanShot 2024-12-12 at 15 15 22@2x

These were the result of treating every `.bz2` file as a tar-based archive. This PR fixes the extraction by directly extracting non-tar.bz2 files:
```
$ go run cmd/mal/mal.go --include-data-files=true analyze ~/Downloads/test1.csv.bz2
🔎 Scanning "~/Downloads/test1.csv.bz2"
```

As opposed to:
```
🔎 Scanning "~/Downloads/test1.csv.bz2"
time=2024-12-12T15:05:24.317-06:00 level=ERROR source=.../repos/chainguard-dev/malcontent/pkg/action/scan.go:330 msg="unable to process ~/Downloads/test1.csv.bz2: extract to temp: failed to extract ~/Downloads/test1.csv.bz2: failed to read tar header: archive/tar: invalid tar header"
```

### CODE REVIEW

File: pkg/action/archive.go
Comment (tstromberg) [inline pkg/action/archive.go:143]: While you are here, `.tbz` is a less common short extension form of `.tar.bz2`. If you add support for it, it might inspire a slightly different refactoring of this code.

File: pkg/action/archive.go
Comment (tstromberg) [inline pkg/action/archive.go:150]: Is world readable necessary? If not, maybe just `0700`?

Comment (tstromberg) [reply pkg/action/archive.go:150]: instead of `filepath.Dir(target)`, there is already `d`; is it sufficient?

File: pkg/action/archive.go
Comment (tstromberg) [inline pkg/action/archive.go:145]: should code be located outside of the `extractTar` method then?

If you made this new code it's own method, it'd remove the complexity.

File: pkg/action/archive.go
Comment (tstromberg) [inline pkg/action/archive.go:154]: Can you add a comment as to what this directive is being used for?

File: pkg/action/archive.go
Comment (tstromberg) [inline pkg/action/archive.go:155]: I can't tell - but is this creating a new file in the same directory as the source file?  That may not work in all circumstances; for example, scanning a read-only mount or local directory where the user has read but not write access.

**Review by tstromberg — CHANGES_REQUESTED** [review]

File: pkg/action/archive.go
Comment (egibs) [reply pkg/action/archive.go:155]: It's landing in `/tmp` unless I'm misunderstanding:
```
time=2024-12-12T15:30:33.982-06:00 level=DEBUG source=.../repos/chainguard-dev/malcontent/pkg/action/archive.go:86 msg="extracting tar" dir=/var/folders/n6/xxn5d2zd3l1gghpx_7qppzs00000gn/T/test1.csv.bz21307193049
```

File: pkg/action/archive.go
Comment (egibs) [reply pkg/action/archive.go]: Added in [`c32840a` (#702)](https://github.com/chainguard-dev/malcontent/pull/702/commits/c32840af4cba55983f37e6e233c37170267ba06b).

File: pkg/action/archive.go
Comment (egibs) [reply pkg/action/archive.go]: Changed in [`c32840a` (#702)](https://github.com/chainguard-dev/malcontent/pull/702/commits/c32840af4cba55983f37e6e233c37170267ba06b).

File: pkg/action/archive.go
Comment (egibs) [reply pkg/action/archive.go]: Updated in [`c32840a` (#702)](https://github.com/chainguard-dev/malcontent/pull/702/commits/c32840af4cba55983f37e6e233c37170267ba06b).

File: pkg/action/archive.go
Comment (egibs) [reply pkg/action/archive.go]: Added in [`c32840a` (#702)](https://github.com/chainguard-dev/malcontent/pull/702/commits/c32840af4cba55983f37e6e233c37170267ba06b).

**Comment by egibs** [comment]
Once this merges I'm going to have a follow up PR to clean up the `archive.go` organization since it's getting pretty lengthy now.

**Pull request merged**
